pages/installing-packages.md

```diff
@@ -0,0 +1,110 @@
+
+# What is "mamba"?
+
+We recommend using "mamba", which is a drop-in replacement for "conda" that is many times faster. You particularly notice it when a package has many dependencies or complex version constraints on its dependencies.
+
+In fact, the conda developers are [incorporating mamba into conda](https://www.anaconda.com/blog/a-faster-conda-for-a-growing-community), though at the time of this writing, that integration is still experimental. These instructions will describe how to use mamba directly.
+
+# Where will the files go?
+
+The entire Python distribution, with all packages and the shared libraries that support them, will go in a new directory, most likely in your home directory and named `mambaforge`. All of the files in it are installed with your own user permissions (i.e. not superuser/`sudo`).
+
+## How to get rid of them if you change your mind
+
+1. Delete that directory with `rm -rf ~/mambaforge`.
+2. Delete a file named `~/.condarc`, if you have one.
+3. Check your shell configuration file, probably named `~/.bashrc`, for a "`>>> conda initialize`" section. If you have one, delete it.
+
+Those three steps will remove any vestige of the conda installation.
+
+## How to save an old package list before deleting it
+
+If you already have a conda installation, you can bundle your current environment into an environment file with
+
+```bash
+conda env export --from-history > old-environment.yml
+```
+
+After reinstalling, you can reconstruct it with
+
+```bash
+conda env create -f old-environment.yml
+```
+
+# Installing Miniforge
+
+Miniforge is [distributed on GitHub](https://github.com/conda-forge/miniforge).
+
+The steps of the installation procedure are (1) download an installer script, (2) run it, and (3) answer interactive prompts.
+
+Of the four combinations Miniforge gives you (conda vs mamba, Python vs PyPy), [we recommend mamba with Python, which is this table](https://github.com/conda-forge/miniforge#mambaforge).
+
+Within each table is a list of architectures. On Mac and Linux, you can get the name of your architecture from
+
+```bash
+uname -i
+```
+
+It is very likely `x86_64`. Select the installation script for your architecture by clicking or right-clicking the link on the Miniforge page.
+
+On Mac or Linux, run the script with
+
+```bash
+bash filename-of-the-script-you-just-downloaded.sh
+```
+
+Windows has a `start` command; see [Miniforge's instructions](https://github.com/conda-forge/miniforge#windows).
+
+The interactive prompts will ask you where you want to install it (default is `~/mambaforge`) and whether you want to have it enabled whenever you start a new terminal or shell (probably yes). Saying "yes" to the latter inserts a "`>>> conda initialize`" section in your shell configuration (probably `~/.bashrc`).
```

Comment:
```suggestion
The interactive prompts will ask you where you want to install it (default is `~/mambaforge`) and whether you want to have it enabled whenever you start a new terminal or shell (probably yes). Saying "yes" to the latter inserts a "`>>> conda initialize`" section in your shell configuration (probably `~/.bashrc`). If you want to have control over your environments (and, specifically, have one environment per project and avoid a "base" environment), you can run `conda config --set auto_activate_base false`, and then manually activate and deactivate environments without replacing access to your system Python when nothing is activated.
```

Reply:
Is this too much information for a simple user? With these instructions, I was targeting users who are not developers, who are not _likely_ to make separate environments even though we recommend it. For them, the base environment with an option to add additional environments is perfect: you get the simple case that people are likely to want to do and remove all barriers to the better base of adding environments when they get sophisticated.

I'd rather leave this additional text out. If they're enough of a power user to want that, I don't think they're reading this page.

Reply:
This was super useful for me to read right now....

Reply:
IMO, it's the most important thing to run as soon as you install conda, and someone working on two projects is not _that_ much of a super user. And nowadays assuming it's okay to take over a user's "python" command everywhere might not be so great - many people entering have previous Python experience, and might not want "python" to suddenly be only this new project they are working on.

Reply:
Maybe another `<details>`? A pet peeve of mine with Conda is its proclivity for modifying your shell config.

Reply:
I like the idea of making it another `<details>`. While we're at it, I like setting `changeps1: false` in my `~/.condarc` and `MAMBA_NO_BANNER=1`. Those can be `<details>`, too, but I wouldn't want to overwhelm a new user with all those suggestions unless they're partially hidden.

I'm not saying that someone working on two projects and keeping them separated by environments is a power user, just that having a "base" and then adding a project-specialized environment later is a gradual introduction to the concept of multiple environments. I believe that at first, users who need this page aren't going to want to hear about all the wonderful things you can do—they just want a set-up that works. The need for environments usually arises later, like when they start that second project, and it's something they can opt into later. The bit of text at the bottom of the page is intended as a seed to plant the idea, and hopefully they remember that something like this can be done when they eventually need it.

I want to minimize the number of things a new user has to think about _at first_. If someone can answer the question, "Do I want a default base environment or opt into every environment?" then they probably don't need this page.

Reply:
There's another point here, in whether conda should override the `python` executable (and other executables with common names) in all new terminals, without user input. It's in a `<detail>`, but that's the focus of the discussion, which might be very relevant for someone who's just sometimes using conda-installed packages.

I put more emphasis in the last section that we encourage the use of isolated environments, but just don't go into a lot of depth about them here, in this "first steps" article.

Reply:
IMO, `changeps1: false` is terrible - it makes you forget that an environment is activated! It's the opposite of `auto_activate_base`; the idea is you "enter" an environment (and your PS1 tells you that) if you want to work, and it doesn't break your system if you don't activate anything.

Also, I think with tutorials and suggested beginner environments, we are providing environment.yml files; users need to know how to go in and out of an environment to use them, though!

Reply:
I use `changeps1: false`, but I didn't recommend that in the document. It just bugs me to have a prompt longer than `% `. (I don't expect the prompt to give me any information other than "a process isn't running.") I end up typing

```bash
conda env list
```

a lot.

Anyway, that's not for beginners. It's just to accommodate me, who got used to seeing it a different way.